File: lib/presentation/pages/instruction_page.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../../core/constants/app_colors.dart';
import '../../core/constants/app_routes.dart';
import '../../core/constants/app_text_styles.dart';
import '../../core/theme/app_theme.dart';
import '../widgets/app_card.dart';

class InstructionPage extends StatelessWidget {
  const InstructionPage({super.key});

  static const List<_InstructionItem> _items = [
    _InstructionItem(
      icon: Icons.smartphone,
      title: 'Attach Your Phone',
      description:
          'Securely fasten your smartphone to your ankle or lower leg using the provided strap.',
    ),
    _InstructionItem(
      icon: Icons.directions_walk,
      title: 'Walk Naturally',
      description:
          'Walk at your normal pace. Do not try to walk differently — natural gait gives the best results.',
    ),
    _InstructionItem(
      icon: Icons.repeat,
      title: 'Record Both Feet',
      description:
          'You will record each foot separately. Complete the right foot first, then the left.',
    ),
    _InstructionItem(
      icon: Icons.health_and_safety_outlined,
      title: 'Stay Safe',
      description:
          'Ensure a clear, flat path of at least 10 meters. Have a support person nearby if needed.',
    ),
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      body: Column(
        children: [
          // App bar
          _BackAppBar(title: 'Test Instructions'),
          // Scrollable content
          Expanded(
            child: SingleChildScrollView(
              padding: const EdgeInsets.all(AppTheme.marginMobile),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  // Hero illustration
                  Container(
                    width: double.infinity,
                    height: 180,
                    decoration: BoxDecoration(
                      gradient: LinearGradient(
                        begin: Alignment.topLeft,
                        end: Alignment.bottomRight,
                        colors: [
                          AppColors.primary.withAlpha(26),
                          AppColors.primaryContainer.withAlpha(26),
                        ],
                      ),
                      borderRadius: AppTheme.radiusLg,
                    ),
                    child: const Icon(
                      Icons.directions_walk,
                      size: 100,
                      color: AppColors.primary,
                    ),
                  ),
                  const SizedBox(height: AppTheme.xl),
                  Text('Before You Begin', style: AppText.headlineSm),
                  const SizedBox(height: AppTheme.md),
                  ..._items.map((item) => Padding(
                        padding:
                            const EdgeInsets.only(bottom: AppTheme.sm),
                        child: _InstructionCard(item: item),
                      )),
                  const SizedBox(height: AppTheme.md),
                  // Reassurance banner
                  Container(
                    padding: const EdgeInsets.all(AppTheme.md),
                    decoration: BoxDecoration(
                      color: AppColors.primaryContainer.withAlpha(26),
                      borderRadius: AppTheme.radiusDefault,
                      border: Border.all(
                          color: AppColors.primaryContainer.withAlpha(80)),
                    ),
                    child: Row(
                      children: [
                        const Icon(Icons.lock_outline,
                            color: AppColors.primary, size: 20),
                        const SizedBox(width: AppTheme.sm),
                        Expanded(
                          child: Text(
                            'Your data is encrypted and stored securely. Only you can access your results.',
                            style: AppText.bodySm,
                          ),
                        ),
                      ],
                    ),
                  ),
                  const SizedBox(height: 100),
                ],
              ),
            ),
          ),
          // Sticky footer
          Container(
            padding: const EdgeInsets.all(AppTheme.marginMobile),
            decoration: BoxDecoration(
              color: AppColors.surface,
              border: Border(
                top: BorderSide(
                    color: AppColors.outlineVariant.withAlpha(51)),
              ),
            ),
            child: SafeArea(
              top: false,
              child: GestureDetector(
                onTap: () => context.go(AppRoutes.footSelection),
                child: Container(
                  width: double.infinity,
                  padding: const EdgeInsets.symmetric(vertical: 16),
                  decoration: BoxDecoration(
                    color: AppColors.primary,
                    borderRadius: AppTheme.radiusFull,
                    boxShadow: AppTheme.primaryShadow,
                  ),
                  child: Text(
                    'Continue',
                    style:
                        AppText.labelMd.copyWith(color: AppColors.onPrimary),
                    textAlign: TextAlign.center,
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _BackAppBar extends StatelessWidget {
  final String title;
  const _BackAppBar({required this.title});

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 64 + MediaQuery.of(context).padding.top,
      padding: EdgeInsets.only(
          top: MediaQuery.of(context).padding.top,
          left: 20,
          right: 20),
      color: AppColors.surface,
      child: Row(
        children: [
          GestureDetector(
            onTap: () => context.go(AppRoutes.home),
            child: Container(
              width: 40,
              height: 40,
              decoration: BoxDecoration(
                color: AppColors.surfaceContainerLow,
                borderRadius: BorderRadius.circular(12),
              ),
              child: const Icon(Icons.arrow_back,
                  size: 20, color: AppColors.onSurface),
            ),
          ),
          const SizedBox(width: 12),
          Text(title, style: AppText.headlineSm),
        ],
      ),
    );
  }
}

class _InstructionItem {
  final IconData icon;
  final String title;
  final String description;
  const _InstructionItem(
      {required this.icon,
      required this.title,
      required this.description});
}

class _InstructionCard extends StatefulWidget {
  final _InstructionItem item;
  const _InstructionCard({required this.item});

  @override
  State<_InstructionCard> createState() => _InstructionCardState();
}

class _InstructionCardState extends State<_InstructionCard> {
  bool _checked = false;

  @override
  Widget build(BuildContext context) {
    return AppCard(
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Container(
            width: 44,
            height: 44,
            decoration: BoxDecoration(
              color: AppColors.primary.withAlpha(26),
              borderRadius: BorderRadius.circular(12),
            ),
            child:
                Icon(widget.item.icon, color: AppColors.primary, size: 22),
          ),
          const SizedBox(width: AppTheme.md),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(widget.item.title, style: AppText.labelMd),
                const SizedBox(height: 4),
                Text(widget.item.description, style: AppText.bodySm),
              ],
            ),
          ),
          const SizedBox(width: AppTheme.sm),
          GestureDetector(
            onTap: () => setState(() => _checked = !_checked),
            child: AnimatedContainer(
              duration: const Duration(milliseconds: 200),
              width: 24,
              height: 24,
              decoration: BoxDecoration(
                color: _checked ? AppColors.primary : Colors.transparent,
                borderRadius: BorderRadius.circular(6),
                border: Border.all(
                  color: _checked
                      ? AppColors.primary
                      : AppColors.outlineVariant,
                  width: 2,
                ),
              ),
              child: _checked
                  ? const Icon(Icons.check,
                      size: 14, color: AppColors.onPrimary)
                  : null,
            ),
          ),
        ],
      ),
    );
  }
}

File: src/presentation/pages/instruction_page.js
import { instructionCard } from '../widgets/instruction_card.js';

const INSTRUCTIONS = [
  { icon: 'smartphone',       title: 'Attach Your Phone',  description: 'Secure your phone firmly on the lateral side of your selected foot using a strap or tight sock.' },
  { icon: 'directions_walk',  title: 'Walk Naturally',     description: 'Walk at your most comfortable, everyday speed across a flat surface for 10 meters.' },
  { icon: 'repeat',           title: 'Record Both Feet',   description: 'To get a full analysis, you will need to repeat the test for both your right and left foot.' },
  { icon: 'health_and_safety', title: 'Stay Safe',          description: 'Ensure your walking path is clear of obstacles and avoid slippery surfaces during the test.' },
];

const HERO_IMG = 'https://lh3.googleusercontent.com/aida-public/AB6AXuDWzQ91i_Cm363zVQEjSxH_-J-noNQNNVH9SJFp4b_XawTRBwIzeWdaJalHrI6gxcvk7iwPKOy43C3rFCaF0b7o8feO_lYsxQqmREkuEJnyWw1yYR_71SBHJID8954iBp_ZvWtR6xw2A9uj-A-OLtKt5WOz5d1AY6DCG8yqFeVM4iUEGwbiwhYfKAUCtb-a6EBLJH7F7nKEUab7LCRnYPygApcSzy081YvIdmLsDN0PwJWb7YiCTHXNHggKi3ukq2EwcpX_f5U_mfA';

export function render() {
  return `
    <div class="bg-background min-h-screen flex flex-col font-body-md text-on-background selection:bg-primary-container selection:text-on-primary-container">
      <!-- Header -->
      <header class="bg-surface/80 backdrop-blur-md sticky top-0 z-50 shadow-sm flex justify-between items-center px-margin-mobile h-16 w-full">
        <div class="flex items-center gap-md">
          <button data-nav-to="/home" class="transition-all duration-200 active:scale-95 text-on-surface-variant">
            <span class="material-symbols-outlined">arrow_back</span>
          </button>
          <h1 class="font-headline-sm text-headline-sm text-primary">Before You Start</h1>
        </div>
        <div class="w-10 h-10 rounded-full bg-surface-container flex items-center justify-center text-primary">
          <span class="material-symbols-outlined">info</span>
        </div>
      </header>

      <main class="flex-grow px-margin-mobile pt-xl pb-xl max-w-xl mx-auto w-full">
        <!-- Hero -->
        <section class="mb-xl text-center">
          <div class="w-full aspect-[16/9] rounded-lg bg-surface-container-high mb-lg overflow-hidden relative">
            <div class="absolute inset-0 bg-gradient-to-tr from-primary/10 to-transparent"></div>
            <img src="${HERO_IMG}" alt="Wellness Guide" class="w-full h-full object-cover">
          </div>
          <p class="font-body-lg text-body-lg text-on-surface-variant max-w-sm mx-auto">
            Follow these simple steps to ensure your gait analysis is accurate and helpful.
          </p>
        </section>

        <!-- Instruction Cards -->
        <div class="space-y-md">
          ${INSTRUCTIONS.map(i => instructionCard(i)).join('')}
        </div>

        <!-- Reassurance Banner -->
        <div class="mt-xl p-lg bg-secondary-container/30 rounded-lg border border-secondary-container flex items-center gap-md">
          <span class="material-symbols-outlined text-on-secondary-container">verified_user</span>
          <p class="font-label-md text-label-md text-on-secondary-container">
            Your data is encrypted and used only for your clinical analysis.
          </p>
        </div>
      </main>

      <!-- Sticky Footer -->
      <footer class="p-margin-mobile bg-surface/90 backdrop-blur-md sticky bottom-0">
        <button id="continue-btn" class="w-full h-14 rounded-full bg-primary text-on-primary font-headline-sm text-headline-sm shadow-lg transition-transform active:scale-95 hover:bg-primary/90">
          Continue
        </button>
      </footer>
    </div>
  `;
}

export function mount(navigate) {
  document.querySelectorAll('[data-nav-to]').forEach(el => {
    el.addEventListener('click', () => navigate(el.dataset.navTo));
  });
  document.getElementById('continue-btn')?.addEventListener('click', () => navigate('/foot-selection'));
}
